app/models/person.rb

```diff
@@ -237,11 +237,15 @@ def name(opts = {})
     if self.profile.nil?
       fix_profile
     end
-    @name ||= Person.name_from_attrs(self.profile.first_name, self.profile.last_name, self.diaspora_handle)
+    @name ||= Person.name_from_attrs(self.profile.full_name, self.diaspora_handle)
   end
 
-  def self.name_from_attrs(first_name, last_name, diaspora_handle)
-    first_name.blank? && last_name.blank? ? diaspora_handle : "#{first_name.to_s.strip} #{last_name.to_s.strip}".strip
+  def full_name
```

Comment:
Delegate is probably better (https://github.com/diaspora/diaspora/blob/develop/app/models/person.rb#L25), because you would need `@invite.try(:profile).try(:full_name).empty?` otherwise.